### PR TITLE
Add a leading space on binary operator token trailing comments

### PR DESCRIPTION
**Pull request by weswigham**
Looking through our RWC results, I realized that we could be a bit better in how we emitted comments on binary expressions (from #16584), thanks to work done in #17557.

### CODE REVIEW

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -1348,7 +1348,7 @@ namespace ts {
             increaseIndentIf(indentBeforeOperator, isCommaOperator ? " " : undefined);
             emitLeadingCommentsOfPosition(node.operatorToken.pos);
             writeTokenNode(node.operatorToken);
-            emitTrailingCommentsOfPosition(node.operatorToken.end);
+            emitTrailingCommentsOfPosition(node.operatorToken.end, /*prefixSpace*/ true); // Comma should have a space after it
```
Comment (sandersn) [inline src/compiler/emitter.ts]: Comma? This is true for all binary operators, right?

File: tests/baselines/reference/commentOnBinaryOperator1.js [hunk 2]
```diff
@@ -21,5 +21,5 @@ var b = 'some'
     + 'text';
 var c = 'some'
     /* comment */
-    +/*comment1*/ 
+    + /*comment1*/
```
Comment (rbuckton) [inline tests/baselines/reference/commentOnBinaryOperator1.js]: This looks incorrect.

File: tests/baselines/reference/parserGreaterThanTokenAmbiguity10.js [hunk 3]
```diff
@@ -7,5 +7,5 @@
 //// [parserGreaterThanTokenAmbiguity10.js]
 1
     // before
-    >>>// after
+    >>> // after
```
Comment (rbuckton) [inline tests/baselines/reference/parserGreaterThanTokenAmbiguity10.js]: We shouldn't emit a space if the comment has a leading newline

File: tests/baselines/reference/commentOnBinaryOperator1.js (hunk 2 above)
Comment (weswigham) [inline tests/baselines/reference/commentOnBinaryOperator1.js]: This output matches the input now.

Comment (weswigham) [inline tests/baselines/reference/commentOnBinaryOperator1.js]: And no longer has trailing whitespace.

File: src/compiler/emitter.ts (hunk 1 above)
Comment (weswigham) [inline src/compiler/emitter.ts]: Yep, that comment's a bit too specific. I'll correct it.

File: tests/baselines/reference/parserGreaterThanTokenAmbiguity10.js (hunk 3 above)
Comment (weswigham) [inline tests/baselines/reference/parserGreaterThanTokenAmbiguity10.js]: This comment doesn't have a leading newline, though? And the output for this test now exactly matches the input.

**Comment by weswigham** [comment]
@rbuckton In all cases, both in our test suite and in our RWC suite, just having there be a space before the trailing comment of a binary operator instead of after _always_ matches the original comment spacing. It's likely just because not many people write code like `var a=/*comment*/ foo;` or `(2+/*comment*/2)`, but we don't preserve whitespace verbatim in many (any?) of these places anyway. This is just a _much_ more common default.

**Comment by rbuckton** [comment]
You are correct. GitHub's diff didn't give enough apparent context and the `+` and `>>>`  looked like custom diff emit from one of our test artifacts. These do look correct. 👍

**Pull request merged**